src/main/groovy/com/bmuschko/gradle/docker/tasks/container/DockerLogsContainer.groovy

```diff
@@ -0,0 +1,150 @@
+     * Limit the number of lines of existing output. This cannot be set if #tailAll is also set.
+     * Default is unspecified (docker defaults to all lines).
+     */
+    @Input
+    @Optional
+    Integer tailCount
+
+    /**
+     * Include standard out.
+     * Default is true.
+     */
+    @Input
+    @Optional
+    boolean stdOut = true
+
+
+    /**
+     * Include standard err.
+     * Default is true.
+     */
+    @Input
+    @Optional
+    boolean stdErr = true
+
+    /**
+     * Set to the true to include a timestamp for each line in the output.
+     * Default is unspecified (docker defaults to false).
+     */
+    @Input
+    @Optional
+    Boolean showTimestamps
+
+    /**
+     * Limit the output to lines on or after the specified date.
+     * Default is unspecified (docker defaults to all lines).
+     */
+    @Input
+    @Optional
+    Date since
+
+    private createLoggingCallback() {
+        Class callbackClass = threadContextClassLoader.loadClass("com.github.dockerjava.core.command.LogContainerResultCallback")
+        def delegate = callbackClass.newInstance()
+
+        Class enhancerClass = threadContextClassLoader.loadClass('net.sf.cglib.proxy.Enhancer')
+        def enhancer = enhancerClass.getConstructor().newInstance()
+        enhancer.setSuperclass(callbackClass)
+        enhancer.setCallback([
+
+            invoke: {Object proxy, java.lang.reflect.Method method, Object[] args ->
+                if ("onNext" == method.name) {
+                  def frame = args[0]
+                  switch (frame.streamType as String) {
+                    case "STDOUT":
+                    case "RAW":
+                        System.out.print(new String(frame.payload))
+                        break
+                    case "STDERR":
+                        System.err.print(new String(frame.payload))
+                        break
+                  }
+                }
+                method.invoke(delegate, args)
+            }
+
+        ].asType(threadContextClassLoader.loadClass('net.sf.cglib.proxy.InvocationHandler')))
+
+        enhancer.create()
+    }
+
+    @Override
+    void runRemoteCommand(dockerClient) {
+        logger.info "Logs for container with ID '${getContainerId()}'."
+        def logCommand = dockerClient.logContainerCmd(getContainerId())
+        setContainerCommandConfig(logCommand)
+        logCommand.exec(createLoggingCallback())?.awaitCompletion()
+    }
+
+    private void setContainerCommandConfig(logsCommand) {
+        if (getFollow() != null) {
+            logsCommand.withFollowStream(getFollow())
+        }
+
+        if (getShowTimestamps() != null) {
+            logsCommand.withTimestamps(getShowTimestamps())
+        }
+
+        logsCommand.withStdOut(getStdOut()).withStdErr(getStdErr())
+
+        if (getTailAll() != null && getTailCount() != null) {
+            throw new InvalidUserDataException("Conflicting parameters: only one of tailAll and tailCount can be specified")
```

Comment:
InvalidUserDataException extends GradleException, so being more specific isn't hurting handling the more generic case, right? According to Gradle API, InvalidUserDataException is appropriate here:
```
A InvalidUserDataException is thrown, if a user is providing illegal data for the build.
```